---
arturl_encode: "68747470733a2f2f626c6f67:2e6373646e2e6e65742f77656978696e5f3330343137343837:2f61727469636c652f64657461696c732f3938343130373831"
layout: post
title: "java.io.IOException-java.io.EOFException-Unexpected-end-of-input-stream错误"
date: 2023-07-21 11:30:48 +08:00
description: "报错现象：Diagnostic Messages for this Task:Error: java"
keywords: "Hadoop,Hive,IOException"
categories: ['未分类']
tags: ['大数据', 'Viewui', 'Javascript', 'Java']
artid: "98410781"
image:
  path: https://api.vvhan.com/api/bing?rand=sj&artid=98410781
  alt: "java.io.IOException-java.io.EOFException-Unexpected-end-of-input-stream错误"
render_with_liquid: false
featuredImage: https://bing.ee123.net/img/rand?artid=98410781
featuredImagePreview: https://bing.ee123.net/img/rand?artid=98410781
---

# java.io.IOException: java.io.EOFException: Unexpected end of input stream错误

报错现象：

Diagnostic Messages for this Task:
  
Error: java.io.IOException: java.io.EOFException: Unexpected end of input stream
  
at org.apache.hadoop.hive.io.HiveIOExceptionHandlerChain.handleRecordReaderNextException(HiveIOExceptionHandlerChain.java:121)
  
at org.apache.hadoop.hive.io.HiveIOExceptionHandlerUtil.handleRecordReaderNextException(HiveIOExceptionHandlerUtil.java:77)
  
at org.apache.hadoop.hive.ql.io.HiveContextAwareRecordReader.doNext(HiveContextAwareRecordReader.java:344)
  
at org.apache.hadoop.hive.ql.io.HiveRecordReader.doNext(HiveRecordReader.java:79)
  
at org.apache.hadoop.hive.ql.io.HiveRecordReader.doNext(HiveRecordReader.java:33)
  
at org.apache.hadoop.hive.ql.io.HiveContextAwareRecordReader.next(HiveContextAwareRecordReader.java:122)
  
at org.apache.hadoop.mapred.MapTask$TrackedRecordReader.moveToNext(MapTask.java:199)
  
at org.apache.hadoop.mapred.MapTask$TrackedRecordReader.next(MapTask.java:185)
  
at org.apache.hadoop.mapred.MapRunner.run(MapRunner.java:52)
  
at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:450)
  
at org.apache.hadoop.mapred.MapTask.run(MapTask.java:343)
  
at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:168)
  
at java.security.AccessController.doPrivileged(Native Method)
  
at javax.security.auth.Subject.doAs(Subject.java:415)
  
at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1642)
  
at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:163)
  
Caused by: java.io.EOFException: Unexpected end of input stream
  
at org.apache.hadoop.io.compress.DecompressorStream.decompress(DecompressorStream.java:145)
  
at org.apache.hadoop.io.compress.DecompressorStream.read(DecompressorStream.java:85)
  
at java.io.InputStream.read(InputStream.java:101)
  
at org.apache.hadoop.util.LineReader.fillBuffer(LineReader.java:180)
  
at org.apache.hadoop.util.LineReader.readDefaultLine(LineReader.java:216)
  
at org.apache.hadoop.util.LineReader.readLine(LineReader.java:174)
  
at org.apache.hadoop.mapred.LineRecordReader.skipUtfByteOrderMark(LineRecordReader.java:206)
  
at org.apache.hadoop.mapred.LineRecordReader.next(LineRecordReader.java:244)
  
at org.apache.hadoop.mapred.LineRecordReader.next(LineRecordReader.java:47)
  
at org.apache.hadoop.hive.ql.io.HiveContextAwareRecordReader.doNext(HiveContextAwareRecordReader.java:339)
  
... 13 more

Container killed by the ApplicationMaster.
  
Container killed on request. Exit code is 143
  
Container exited with a non-zero exit code 143

FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask

原因：

hdfs 文件异常造成；

解决：

定位failed attempt，找到处理报错的file;

使用zcat进行测试报错：

![](//images2015.cnblogs.com/blog/359231/201608/359231-20160801125954325-1736686615.png)

删除报错file解决问题；

REF:http://caiguangguang.blog.51cto.com/1652935/1436077/

转载于:https://www.cnblogs.com/chao1118/p/5725434.html